src/App.tsx

```diff
@@ -1,61 +1,119 @@
 import './App.scss';
+import { useState } from 'react';
+import { TodoList } from './components/TodoList';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+function getUser(userId: number): User | null {
+  const findUser = usersFromServer.find(user => user.id === userId);
+
+  return findUser || null;
+}
+
+const todos = todosFromServer.map((todo) => {
+  return (
+    {
+      ...todo,
+      user: getUser(todo.userId),
+    }
+  );
+});
 
 export const App = () => {
+  const [title, setTitle] = useState('');
+  const [selectedUser, chooseUser] = useState('0');
+  const [newTodos, addTodo] = useState(todos);
+  const [isSubmited, changeSubmited] = useState(true);
+
+  const createId = (lastTodos: Todo[]) => {
+    const allIdes = lastTodos.map(todo => {
+      return (todo.id);
+    });
+
+    return Math.max(...allIdes) + 1;
+  };
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/users" method="POST">
+      <form
+        action="/api/users"
+        method="POST"
+        onSubmit={(event) => {
+          event.preventDefault();
+
+          if (+selectedUser > 0 && title.length > 0) {
+            addTodo((prevTodos) => ([
+              ...prevTodos,
+              {
+                id: createId(prevTodos),
+                title,
+                completed: false,
+                userId: +selectedUser,
+                user: getUser(+selectedUser),
+              },
+            ]));
+            setTitle('');
+            chooseUser('0');
+            changeSubmited(true);
+          } else {
+            changeSubmited(false);
+          }
+        }}
+      >
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          <input
+            type="text"
+            data-cy="titleInput"
+            placeholder="Please enter a title"
+            value={title}
+            onChange={(event) => {
+              return setTitle(event.target.value);
```

Comment:
no need to return the function here
```suggestion
              setTitle(event.target.value);
```